### PR TITLE
Bump version metadata to produce correct tester and nightly build precedence

### PR DESCRIPTION
**Pull request by per1234**
### Motivation

On every startup, the Arduino IDE checks for new versions of the IDE (https://github.com/arduino/arduino-ide/pull/797). If a newer version is available, a notification/dialog is shown offering an update.

"Newer" is determined by comparing the version of the user's IDE to the latest available version on the update channel. This comparison is done according to [semver](https://semver.org/).

In order to facilitate beta testing, builds are generated of the Arduino IDE at the current stage in development. These builds are given an identifying version of the following form:

- `<version>-snapshot-<short hash>` - builds generated for every push and pull request that modifies relevant files
- `<version>-nightly-<YYYYMMDD>` - daily builds of the tip of the default branch

The established project management practices cause the `<version>` component of these to be the version of the most recent release.

During the pre-release phase of the project development, all releases had [a pre-release suffix](https://semver.org/#spec-item-9) (e.g., `2.0.0-rc9.4`). Appending the "snapshot" or "nightly" suffix to that pre-release version caused these builds to have the correct precedence (e.g., `2.0.0-rc9.2.snapshot-20cc34c` > `2.0.0-rc9.2`). This situation has changed now that the project is using production release versions (e.g., `2.0.0-nightly-20220915` < `2.0.0`). This causes users of "snapshot" or "nightly" builds to be presented with a spurious update notification on startup (https://github.com/arduino/arduino-ide/issues/1440).

### Change description

The solution is to do a minor bump of the version metadata after creating the release tag.

- Bump the `version` field in the project's `package.json` metadata files to `2.0.1`
- Update [the release procedure documentation](https://github.com/arduino/arduino-ide/blob/main/docs/internal/release-procedure.md) to include this post-tag version bump as a step in the release process

### Other information

The rendered release procedure document can be previewed here:

https://github.com/per1234/arduino-ide/blob/bump-after-release/docs/internal/release-procedure.md#readme

---

Fixes https://github.com/arduino/arduino-ide/issues/1440

---

The pull request also contains an unrelated fix to the link to the ["**Issue Report Guide**"](https://github.com/arduino/arduino-ide/blob/main/docs/contributor-guide/issues.md) from the [issue template chooser page](https://github.com/arduino/arduino-ide/issues/new/choose), which was previously broken.

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: docs/internal/release-procedure.md
Comment (kittaakos) [inline docs/internal/release-procedure.md:31]: IDE2 never followed semver, and does not expose any public APIs, but we can leave this in the doc. In the end, the desired version is a marketing-only thing.

File: docs/internal/release-procedure.md
Comment (kittaakos) [inline docs/internal/release-procedure.md:51]: In the long run, we should use `yarn` to generate these pre-release version numbers instead of manually bumping them. Once this is merged, I will create a follow-up with the request.

Ref: https://classic.yarnpkg.com/en/docs/cli/version/#toc-yarn-version-prerelease

File: docs/internal/release-procedure.md
Comment (kittaakos) [inline docs/internal/release-procedure.md:131]: Assuming you have set the remote to `origin`.

**Review by kittaakos — APPROVED** [review]
Thank you! It's an excellent solution for the time being.

File: docs/internal/release-procedure.md
Comment (per1234) [reply docs/internal/release-procedure.md:31]: Notice that I put "API" in quotes. The spirit of SemVer can still be applied even to projects which don't have a traditional API. The idea of SemVer is that the version number conveys meaning to the users. It is an unfortunate fact that version numbers are sometimes manipulated for marketing purposes, but this is documentation for maintainers, not marketers.

Most often, the choice of version number will be at the discretion of the maintainer and that maintainer should base the choice purely on technical considerations.

File: docs/internal/release-procedure.md
Comment (per1234) [reply docs/internal/release-procedure.md:131]: This pull request only moved the existing content from one place in the document to another.

**Pull request merged**
